### ISSUE TITLE
who broke `spell_data`?

### ISSUE THREAD

**Issue by GuardianDll** [CleverRaven/Cataclysm-DDA#70832]
### Describe the bug

`tr_portal_dungeon_next_level`, trap, used to advance you to the next level in portal dungeon, doesn't work anymore

### Attach save file

[DEBUGDEBUG-trimmed.tar.gz](https://github.com/CleverRaven/Cataclysm-DDA/files/13894136/DEBUGDEBUG-trimmed.tar.gz)

### Steps to reproduce

Debug spawn portal storm
teleport to blue vortex
step on it
nothing happens

### Expected behavior

you teleport to green vortex

### Screenshots

![GIF 10-01-2024 22-22-37](https://github.com/CleverRaven/Cataclysm-DDA/assets/67688115/6807a412-72f9-4739-b909-fb58212d12f2)

### Versions and configuration

- OS: Windows
    - OS Version: 10.0.19045.3803 (22H2)
- Game Version: 2a67568 [64-bit]
- Graphics Version: Tiles
- Game Language: English [en]
- Mods loaded: [
    Dark Days Ahead [dda],
    Disable NPC Needs [no_npc_food],
    Slowdown Fungal Growth [no_fungal_growth],
    Bionic Professions [package_bionic_professions],
    SpeedyDex [speedydex],
    Stats Through Kills [stats_through_kills],
    Stats Through Skills [StatsThroughSkills],
    Magiclysm [magiclysm],
    Xedra Evolved [xedra_evolved]
]

**Comment by andrei8l** [comment]
/confirmed broken by #70479

This check is flipped
```diff
diff --git a/src/map.cpp b/src/map.cpp
index 34f4a70cd1..3d0d83a2ac 100644
--- a/src/map.cpp
+++ b/src/map.cpp
@@ -9715,7 +9715,7 @@ void map::creature_on_trap( Creature &c, const bool may_avoid ) const
     // gliding or boarded in a vehicle means the player is above the trap
     // like a flying monster and can never trigger the trap.
     const Character *const you = c.as_character();
-    if( you != nullptr && ( you->in_vehicle || !you->has_effect( effect_gliding ) ) ) {
+    if( you != nullptr && ( you->in_vehicle || you->has_effect( effect_gliding ) ) ) {
         return;
     }
 
```

In addition, tr_at is wrong (though unrelated)
```diff
diff --git a/src/map.cpp b/src/map.cpp
index 34f4a70cd1..437547d95c 100644
--- a/src/map.cpp
+++ b/src/map.cpp
@@ -6099,7 +6099,7 @@ const trap &map::tr_at( const tripoint &p ) const
 
 const trap &map::tr_at( const tripoint_abs_ms &p ) const
 {
-    return tr_at( p.raw() );
+    return tr_at( bub_from_abs( p ) );
 }
 
 const trap &map::tr_at( const tripoint_bub_ms &p ) const
```

@fairyarmadillo

**Comment by Yobrocharlie** [comment]
Traveling through the green vortex doesn’t work either fun fact your just stuck there

**Comment by GuardianDll** [comment]
fixed by #70836
